### PR TITLE
DOCS-5958 add SignInFirstFactor and SignInSecondFactor to JS type references

### PR DESCRIPTION
**Pull request by alexisintech**
Adds [these resources](https://github.com/clerk/javascript/blob/release/v4/packages/types/src/signIn.ts#L118-#L129) to the docs

### CODE REVIEW

File: docs/references/javascript/sign-in/sign-in.mdx [hunk 1]
```diff
@@ -21,14 +21,13 @@ The `SignIn` object's properties can be split into logical groups, with each gro
...
+| `supportedFirstFactors` | `SignInFirstFactor[]` | Array of the first factors that are supported in the current sign-in. Each factor contains information about the verification strategy that can be used. See the [`SignInFirstFactor`](/docs/references/javascript/types/sign-in-first-factor) type reference for more information. |
+| `supportedSecondFactors` | `SignInSecondFactor[]` | Array of the second factors that are supported in the current sign-in. Each factor contains information about the verification strategy that can be used. This property is populated only when the first factor is verified. See the [`SignInSecondFactor`](/docs/references/javascript/types/sign-in-second-factor) type reference for more information. |
```
Comment (S3Prototype) [inline docs/references/javascript/sign-in/sign-in.mdx:28]:
```suggestion
| `supportedSecondFactors` | [`SignInSecondFactor[]`](/docs/references/javascript/types/sign-in-second-factor)  | Array of the second factors that are supported in the current sign-in. Each factor contains information about the verification strategy that can be used. This property is populated only when the first factor is verified. See the [`SignInSecondFactor`](/docs/references/javascript/types/sign-in-second-factor) type reference for more information. |
```

File: docs/references/javascript/sign-in/sign-in.mdx [hunk 2]
```diff
@@ -21,14 +21,13 @@ The `SignIn` object's properties can be split into logical groups, with each gro
...
+| `firstFactorVerification` | `Verification` | The state of the verification process for the selected first factor. Initially, this property contains an empty verification object, since there is no first factor selected. You need to call the [`prepareFirstFactor`](/docs/references/javascript/sign-in/first-factor#prepare-first-factor) method in order to start the verification process. |
```
Comment (S3Prototype) [inline docs/references/javascript/sign-in/sign-in.mdx:29]: Random comment not related to this line: We need to add the two new pages to `manifest.json`, no?

File: docs/references/javascript/types/sign-in-first-factor.mdx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+title: SignInFirstFactor
+description: The SignInFirstFactor type represents the first factor verification strategy that can be used in the sign-in process.
+---
+
+# `SignInFirstFactor`
+
+The `SignInFirstFactor` type represents the first factor verification strategy that can be used in the sign-in process.
+
+```ts
+type SignInFirstFactor = EmailCodeFactor | EmailLinkFactor | PhoneCodeFactor | PasswordFactor | ResetPasswordPhoneCodeFactor | ResetPasswordEmailCodeFactor | Web3SignatureFactor | OauthFactor | SamlFactor;
```
Comment (S3Prototype) [inline docs/references/javascript/types/sign-in-first-factor.mdx:11]:
```suggestion
type SignInFirstFactor = 
  EmailCodeFactor |
  EmailLinkFactor |
  PhoneCodeFactor |
  PasswordFactor |
  ResetPasswordPhoneCodeFactor |
  ResetPasswordEmailCodeFactor |
  Web3SignatureFactor |
  OauthFactor |
  SamlFactor;
```
I think this is easier to read without scrolling

File: docs/references/javascript/types/sign-in-second-factor.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+title: SignInSecondFactor
+description: The SignInSecondFactor type represents the second factor verification strategy that can be used in the sign-in process.
+---
+
+# `SignInSecondFactor`
+
+The `SignInSecondFactor` type represents the second factor verification strategy that can be used in the sign-in process.
+
+```ts
+type SignInSecondFactor = PhoneCodeFactor | TOTPFactor | BackupCodeFactor;
```
Comment (S3Prototype) [inline docs/references/javascript/types/sign-in-second-factor.mdx:11]:
```suggestion
type SignInSecondFactor =
  PhoneCodeFactor |
  TOTPFactor |
  BackupCodeFactor;
```

**Review by S3Prototype — CHANGES_REQUESTED** [review]
Looks good, but needs the two new pages to be added to `manifest.json` I believe

File: docs/references/javascript/sign-in/sign-in.mdx (hunk 2 above)
Comment (alexisintech) [reply docs/references/javascript/sign-in/sign-in.mdx:29]: WOOPS! yes :P

**Review by S3Prototype — APPROVED** [review]

**Pull request merged**